include/heap.h

```diff
@@ -3,81 +3,105 @@
    See LICENSE and COPYING for usage.
 
  *********************************************************************/
+#define MAX_BLOCK_SIZE 1600
+#define MAX_BLOCK_COUNT 16
 
 #define DECLARE_HEAP(type, orderby) \
     struct heap_ ## type {   \
         uint32_t size;    \
         uint32_t n;       \
-        type *top;        \
+        type *top[MAX_BLOCK_COUNT];        \
     }; \
     typedef struct heap_ ## type heap_ ## type; \
+    static inline type* getElement(struct heap_ ## type *heap, uint32_t idx) \
```

Comment:
also, the mixedCase in `getElement` seems like a coding style from another project. Should be `heap_get_element` probably.